### DISCUSSION TITLE
Invalidate Query not triggering a refetch

### DISCUSSION THREAD

**Discussion by rampantslug** [TanStack/query#8446] [Q&A]
I have a couple of other RN apps that use invalidateQueries successfully and despite copying that code verbatim into my current project it doesnt seem to trigger the refetch on the same screen (I have to navigate away and then back again to see the change).
I even dropped Query version to match the other project that was working fine.

Here are my query and mutation hooks
```
export const useMyRegularAvailabilitiesQuery = () => {
  return useQuery({
    queryKey: ["my-regular-availabilities"],
    queryFn: () => myRegularAvailabilities(),
  })
}

export const useUpdateMyRegularAvailabilitiesMutation = () => {
  const queryClient = useQueryClient()

  return useMutation({
    mutationFn: ({ id, timeslots }: { id: number; timeslots: number[] }) =>
      updateMyRegularAvailabilities({ id, timeslots }),
    onSuccess: () => {
      return queryClient.invalidateQueries({
        queryKey: ["my-regular-availabilities"], refetchType: 'all'
      })
    },
  })
}
```

Here is a simple screen that I am using them in
```
const AvailabilityDaysScreen = () => {

  const { data: regularAvail } = useMyRegularAvailabilitiesQuery()
  const { mutateAsync: updateDayAvail } = useUpdateMyRegularAvailabilitiesMutation()
  
  console.log("regularAvail", regularAvail)

  const doRefetchTest = useCallback(async () => {
    try {
      const result = await updateDayAvail({ id: 238, timeslots: [106] })
    } catch (error) {
      console.log("ERRORO", error)
    }
  }, [regularAvail])

  return (
    <ScreenContainer headerLabel="Availability">
      <Div px="w16">
        <WText variant="h4" my="w24">
          Availability
        </WText>
        {regularAvail ? (
          regularAvail.map((day) => <DayItem key={day.id} day={day} />)
        ) : (
          <WText variant="p2-regular">Loading...</WText>
        )}
        <WButton text="test refetch" onPress={doRefetchTest} />
      </Div>
    </ScreenContainer>
  )
}
```

If I test using my temp button there it WILL update the backend data and call the onSuccess in the mutation but the query wont fire again until I navigate away from the page and then back again.
I was previously doing the update in another page and then routing back to this one and it wouldnt update at all. But putting both the mutation and the query in this page to simplify it and closing can get the update to happen if I close and then reopen this screen.

Any thoughts, or obvious things that I might be doing wrong?

**Comment by eneeio-nm** [comment]
I can't be certain it'll fix the problem, but I notice that you are returning `queryClient.invalidateQueries(...)` within the `onSuccess` hook.

From the [docs](https://tanstack.com/query/latest/docs/framework/react/guides/invalidations-from-mutations) there's no need to `return queryClient.invalidateQueries(...)`, it's just `queryClient.invalidateQueries(...)`

Here's the example code
```ts
import { useMutation, useQueryClient } from '@tanstack/react-query'

const queryClient = useQueryClient()

// When this mutation succeeds, invalidate any queries with the `todos` or `reminders` query key
const mutation = useMutation({
  mutationFn: addTodo,
  onSuccess: () => {
    queryClient.invalidateQueries({ queryKey: ['todos'] })
    queryClient.invalidateQueries({ queryKey: ['reminders'] })
  },
})
```
